app/scripts/modules/core/delivery/executionGroup/executionGroup.component.ts

```diff
@@ -0,0 +1,159 @@
+import {IComponentController, IComponentOptions, IPromise, ITimeoutService, IScope, module} from 'angular';
+import {IModalService} from 'angular-ui-bootstrap';
+import {find, flatten, uniq} from 'lodash';
+
+import {Application} from 'core/application/application.model';
+import {IExecutionDetailsStateParams} from '../delivery.states';
+import {PIPELINE_CONFIG_SERVICE, PipelineConfigService} from 'core/pipeline/config/services/pipelineConfig.service';
+
+import './executionGroup.less';
+
+interface IViewState {
+  triggeringExecution: boolean;
+  open: boolean;
+  poll: IPromise<any>;
+  canTriggerPipelineManually: boolean;
+  canConfigure: boolean;
+  showAccounts: boolean;
+};
+
+export class ExecutionGroupController implements IComponentController {
+  public group: any;
+  public application: Application;
+
+  public deploymentAccounts: string[];
+  public viewState: IViewState;
+  public pipelineConfig: any;
+  public strategyConfig: any;
+
+  static get $inject(): string[] { return ['$scope', '$timeout', '$state', '$stateParams', '$uibModal', 'executionService', 'collapsibleSectionStateCache', 'ExecutionFilterModel', 'pipelineConfigService']; }
+
+  constructor(public $scope: IScope,
+              private $timeout: ITimeoutService,
+              private $state: angular.ui.IStateService,
+              private $stateParams: IExecutionDetailsStateParams,
+              private $uibModal: IModalService,
+              private executionService: any,
+              private collapsibleSectionStateCache: any,
+              private ExecutionFilterModel: any,
+              private pipelineConfigService: PipelineConfigService) {}
+
+  public $onInit(): void {
+    this.deploymentAccounts = uniq(flatten<string>(this.group.executions.map((e: any) => e.deploymentTargets))).sort();
+
+    this.pipelineConfig = find(this.application.pipelineConfigs.data, { name: this.group.heading });
+    this.strategyConfig = find(this.application.strategyConfigs.data, { name: this.group.heading });
+
+    this.viewState = {
+      triggeringExecution: false,
+      open: this.isShowingDetails() || !this.collapsibleSectionStateCache.isSet(this.getSectionCacheKey()) || this.collapsibleSectionStateCache.isExpanded(this.getSectionCacheKey()),
+      poll: null,
+      canTriggerPipelineManually: this.pipelineConfig,
+      canConfigure: this.pipelineConfig || this.strategyConfig,
+      showAccounts: this.ExecutionFilterModel.sortFilter.groupBy === 'name',
+    };
+
+    this.$scope.$on('toggle-expansion', (_, expanded) => {
```

Comment:
Is `_` a convention we're picking up from somewhere for ignored arguments? I only ask because my first thought reading this was "how is lodash getting passed around as an argument?"